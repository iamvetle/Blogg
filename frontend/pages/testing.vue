<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>