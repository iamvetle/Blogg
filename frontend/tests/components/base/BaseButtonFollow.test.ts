import BaseButtonFollow from '~/components/base/BaseButtonFollow.vue';
import { shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

let loggedInUserStore:any;
let wrapper:any;
let pinia = createTestingPinia()

describe('testing basefollowbutton', () => {

    beforeEach(() => {
    loggedInUserStore = useLoggedInUserStore(pinia)


        wrapper = shallowMount(BaseButtonFollow, {
            global: {
                plugins: [pinia],
                components: {},
                mocks: {},
                stubs: {}
            },
            props: {},

        });
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
    });

    test('component exists', () => {
        expect(wrapper.exists()).toBe(true)
    })
    test('This snapchot works?', () => {
        expect(wrapper.html()).toMatchSnapshot()      
    })
});