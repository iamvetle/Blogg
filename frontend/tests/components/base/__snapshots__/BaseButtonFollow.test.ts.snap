// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing basefollowbutton > This snapchot works? 1`] = `
"<div data-test=\\"default-slot\\">
  <!--v-if-->
  <div class=\\"w-fit h-fit cursor-pointer p-1 rounded-md bg-inversePrimary text-onPrimary hover:bg-primary\\" id=\\"follow\\">
    <p class=\\"not-prose\\">Follow</p>
  </div>
</div>"
`;
